src/SparkFun_AS7331.cpp

```diff
@@ -0,0 +1,1051 @@
+#include "SparkFun_AS7331.h"
+
+bool SfeAS7331Driver::begin(const uint8_t &deviceAddress, sfeTkIBus *theBus)
+{
+    // Nullptr check.
+    if (!_theBus && !theBus)
+        return false;
+
+    // Set the internal bus pointer, overriding current bus if it exists.
+    if (theBus != nullptr)
+        setCommunicationBus(theBus);
+
+    // If the address passed in isn't the default, set the new address.
+    if (kDefaultAS7331Addr != deviceAddress)
+        setDeviceAddress(deviceAddress);
+
+    // Get the device setup and ready.
+    return runDefaultSetup();
+}
+
+uint8_t SfeAS7331Driver::getDeviceID(void)
+{
+    // Config registers are only available in Configuration mode.
+    if (!_theBus || _opMode != DEVICE_MODE_CFG)
+        return 0;
```

Comment:
I could see that being helpful. It will be a bit more involved since on reset the default internal state could be different than the device's actual state. I agree that this is a special case and I can work around that.